pkg/agent/controller/agent.go

```diff
@@ -228,94 +304,167 @@ func (a *Controller) serviceImportLister(transform func(si *mcsv1a1.ServiceImpor
+	if !isSupportedServiceType {
+		a.handleServiceExportTransformError(err, localServiceExport,
+			"UnsupportedServiceType", fmt.Sprintf("Service of type %v not supported", svc.Spec.Type))
 
-	svcType, ok := getServiceImportType(svc)
+		return nil, false
+	}
 
-	if !ok {
-		a.updateExportedServiceStatus(svcExport.Name, svcExport.Namespace, corev1.ConditionFalse, invalidServiceType,
-			fmt.Sprintf("Service of type %v not supported", svc.Spec.Type))
-		klog.Errorf("Service type %q not supported", svc.Spec.Type)
+	exportSpec, err := mcs.NewExportSpec(svc, localServiceExport, a.clusterID)
+	if err != nil {
+		a.handleServiceExportTransformError(err, localServiceExport,
+			"ExportSpecCreationFailed", "Error creating service export spec")
 
 		return nil, false
 	}
 
-	serviceImport := a.newServiceImport(svcExport.Name, svcExport.Namespace)
+	err = exportSpec.MarshalObjectMeta(&brokerServiceExport.ObjectMeta)
+	if err != nil {
+		a.handleServiceExportTransformError(err, localServiceExport,
+			"ExportSpecMarshalFailed", "Error marshaling service export spec to ObjectMeta")
 
-	serviceImport.Spec = mcsv1a1.ServiceImportSpec{
-		Ports:                 []mcsv1a1.ServicePort{},
-		Type:                  svcType,
-		SessionAffinityConfig: new(corev1.SessionAffinityConfig),
+		return nil, false
 	}
 
-	serviceImport.Status = mcsv1a1.ServiceImportStatus{
-		Clusters: []mcsv1a1.ClusterStatus{
-			{
-				Cluster: a.clusterID,
-			},
-		},
+	//serviceImport := a.newServiceImport(localServiceExport.Name, localServiceExport.Namespace)
+
+	//serviceImport.Spec = mcsv1a1.ServiceImportSpec{
+	//	Ports:                 []mcsv1a1.ServicePort{},
+	//	Type:                  svcType,
+	//	SessionAffinityConfig: new(corev1.SessionAffinityConfig),
+	//}
+	//
+	//serviceImport.Status = mcsv1a1.ServiceImportStatus{
+	//	Clusters: []mcsv1a1.ClusterStatus{
+	//		{
+	//			Cluster: a.clusterID,
+	//		},
+	//	},
+	//}
+
+	//if svcType == mcsv1a1.ClusterSetIP {
+	//	if a.globalnetEnabled {
+	//		ip, reason, msg := a.getGlobalIP(svc)
+	//		if ip == "" {
+	//			klog.V(log.DEBUG).Infof("Service to be exported (%s/%s) doesn't have a global IP yet", localServiceExport.Namespace, localServiceExport.Name)
+	//			// Globalnet enabled but serviceObj doesn't have globalIp yet, Update the status and requeue
+	//			a.updateExportedServiceStatus(localServiceExport.Name, localServiceExport.Namespace, corev1.ConditionFalse, reason, msg)
+	//
+	//			return nil, true
+	//		}
+	//
+	//		serviceImport.Spec.IPs = []string{ip}
+	//	} else {
+	//		serviceImport.Spec.IPs = []string{svc.Spec.ClusterIP}
+	//	}
+	//
+	//	serviceImport.Spec.Ports = a.getPortsForService(svc)
+	//	/* We also store the clusterIP in an annotation as an optimization to recover it in case the IPs are
+	//	cleared out when here's no backing Endpoint pods.
+	//	*/
+	//	serviceImport.Annotations[clusterIP] = serviceImport.Spec.IPs[0]
+	//}
+
+	a.updateExportedServiceStatus(localServiceExport.Name, localServiceExport.Namespace,
```

Comment:
it's not valid until sync'ed? I thought there was another condition for that (Synced).
Also, do we want a "status synced from broker" as well or is it overkill (since reflected by creation of status Conflict)?
Be aware that k8s metav1.Condition defined True, False and Unknown as base status' states. For reference...

Reply:
lh currently assuemes its not valid until synced